clash-prelude/src/Clash/Class/BitPack/Internal.hs

```diff
@@ -1,102 +1,441 @@
 {-|
-Copyright  :  (C) 2019, QBayLogic B.V.
+Copyright  :  (C) 2013-2016, University of Twente,
+                  2016-2017, Myrtle Software Ltd
 License    :  BSD2 (see the file LICENSE)
 Maintainer :  Christiaan Baaij <christiaan.baaij@gmail.com>
 -}
+
 {-# LANGUAGE CPP #-}
+{-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+{-# LANGUAGE Trustworthy #-}
+
+{-# OPTIONS_GHC -fplugin=GHC.TypeLits.Extra.Solver #-}
+{-# OPTIONS_GHC -fplugin=GHC.TypeLits.Normalise #-}
+{-# OPTIONS_GHC -fplugin=GHC.TypeLits.KnownNat.Solver #-}
+{-# OPTIONS_HADDOCK show-extensions #-}
+
+#include "MachDeps.h"
+
+module Clash.Class.BitPack.Internal where
+
+import Control.Exception              (catch, evaluate)
+import Data.Binary.IEEE754            (doubleToWord, floatToWord, wordToDouble,
+                                       wordToFloat)
+
+import Data.Complex                   (Complex)
+import Data.Int
+import Data.Ord                       (Down)
+import Data.Word
+import Foreign.C.Types                (CUShort)
+import GHC.TypeLits                   (KnownNat, Nat, type (+), type (-))
+import Numeric.Half                   (Half (..))
+import GHC.Generics
+import GHC.TypeLits.Extra             (CLog, Max)
+import Prelude                        hiding (map)
+import System.IO.Unsafe               (unsafeDupablePerformIO)
+
+import Clash.Promoted.Nat             (SNat(..), snatToNum)
+import Clash.Annotations.Primitive    (hasBlackBox)
+import Clash.Class.BitPack.Internal.TH (deriveBitPackTuples)
+import Clash.Class.Resize             (zeroExtend, resize)
+import Clash.Sized.Internal.BitVector
+  (BitVector, Bit, pack#, split#, checkUnpackUndef, undefined#, unpack#,
+   unsafeToNatural, (++#), isLike#)
```

Comment:
another nitpick: I feel if the diff is going to be all green anyway we may as well drop the padding before the import lists. Saves them from being partially out of line / all changed in a big diff later

Reply:
Also alphabetize it!

Reply:
I feel the padding makes it more readable, even if it isn't all perfectly aligned.

Reply:
I do too, but there are some files where it just gets out of sync. We kind of have a weird mixup of styles for imports in different modules and they all end up getting messy, so I would prefer picking the least-easily messed up option

Reply:
And things like adding a qualified module and having to indent all the other import names, or aligning the import lists then having to realign just makes diffs noisier. We avoid this in other places by not aligning things like alternatives in case expressions (which also makes it less readable IMO) so it seems weird to take that stance in some places and not others

We could also leave it for now and bikeshed it later. I do think we should be more consistent with style though

Reply:
https://gist.github.com/martijnbastiaan/26be40480c2edcc216ee28d0191bcda7 <= best style IMO (until `qualified` can be used in the post position on all supported ghcs)

Reply:
which would make it look like https://gist.github.com/martijnbastiaan/8245012dfe0b6a11dca36454ad741c0a

Reply:
What I dislike most about alignment is that it's not forced, so you end up with garbage-looking imports over time. I wouldn't mind tools enforcing it, but not as a separate compile step. If we're going to do that it needs to be something [like this](https://github.com/ocharles/hlint-source-plugin).

Reply:
For the time being I'm leaving this alone. I would like to change it but holding the PR back while we bikeshed style seems counterproductive. I have alphabetised the imports though, and moved the `import Prelude` to the start because it's somewhat "special" as an import